pkg/cmos/cmos.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2012-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+// Package cmos lets you read and write to cmos registers while doing basic checks on valid register selections.
+
+package cmos
+
+import (
+	"fmt"
+
+	"github.com/u-root/u-root/pkg/memio"
+)
+
+const (
+	cmosRegPort  = 0x70
+	cmosDataPort = 0x71
+)
+
+// Read reads a register reg from CMOS into data.
+func Read(reg int64, data memio.UintN) error {
+	regVal := memio.Uint8(reg)
+	if regVal%128 < 14 {
+		return fmt.Errorf("byte %d is inside the range 0-13 which is reserved for RTC", regVal)
+	}
+	if err := memio.Out(cmosRegPort, &regVal); err != nil {
+		return err
+	}
+	return memio.In(cmosDataPort, data)
+}
+
+// Write writes value data into CMOS register reg.
+func Write(reg int64, data memio.UintN) error {
+	regVal := memio.Uint8(reg)
+	if regVal%128 < 14 {
+		return fmt.Errorf("byte %d is inside the range 0-13 which is reserved for RTC", regVal)
```

Comment:
Why are we not allowing RTC?

Reply:
Our internal tool for modifying the cmos doesn't allow it. I'm not too sure what the impacts of changing the rtc bytes are so if nobody has any issues with it I can remove the check.

Reply:
Alternatively, I could add another function to cmds/core/io/cmos.go for read and write and perform the checks there. That way the pkg functions can be more flexible for future references while preserving the checks for the command.

Reply:
or you could add a rtcr and rtcw command which explicitly writes the rtc, i.e. only allows the range 0 to 13?